src/main/java/com/twilio/oai/api/JavaApiResourceBuilder.java

```diff
@@ -358,8 +358,21 @@ private ArrayList<List<CodegenParameter>> generateSignatureList(final CodegenOpe
             }
         }
         conditionalCodegenParam = Lists.cartesianProduct(conditionalCodegenParam);
-        ArrayList<List<CodegenParameter>> signatureList = new ArrayList<>();
+        List<List<CodegenParameter>> filteredConditionalCodegenParam = new ArrayList<>();
+        //Remove duplicates from signatureList
+        ArrayList<List<String>> signatureHashSet = new ArrayList<>();
         for(List<CodegenParameter> paramList : conditionalCodegenParam){
+            List<String> hash = new ArrayList<>();
+            for(CodegenParameter param : paramList){
+                hash.add(param.dataType);
```

Comment:
nit: consider using streams and lambda to convert List<CodegenParameter> to List<String>.